core/src/main/java/nl/nn/adapterframework/doc/model/FrankDocModel.java

```diff
@@ -1,45 +1,143 @@
 package nl.nn.adapterframework.doc.model;
 
+import java.io.IOException;
 import java.lang.reflect.Method;
+import java.text.ParseException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.StringTokenizer;
 import java.util.stream.Collectors;
 
+import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.Logger;
 import org.springframework.core.annotation.AnnotationUtils;
+import org.xml.sax.SAXException;
 
 import lombok.Getter;
 import lombok.Setter;
+import nl.nn.adapterframework.configuration.digester.DigesterRule;
+import nl.nn.adapterframework.configuration.digester.DigesterRulesHandler;
+import nl.nn.adapterframework.core.Resource;
 import nl.nn.adapterframework.doc.IbisDoc;
 import nl.nn.adapterframework.doc.IbisDocRef;
+import nl.nn.adapterframework.doc.Utils;
 import nl.nn.adapterframework.util.LogUtil;
+import nl.nn.adapterframework.util.XmlUtils;
 
 public class FrankDocModel {
-	private static final String JAVA_STRING = "java.lang.String";
-
 	private static Logger log = LogUtil.getLogger(FrankDocModel.class);
-
+	private static final String DIGESTER_RULES = "digester-rules.xml";
+	
+	private @Getter Map<String, ConfigChildSetterDescriptor> configChildDescriptors;
 	private @Getter List<FrankDocGroup> groups;
 	private @Getter Map<String, FrankElement> allElements = new HashMap<>();
+	private @Getter Map<String, ElementType> allTypes = new HashMap<>();
+
+	/**
+	 * Get the FrankDocModel needed in production. This is just a first draft. The
+	 * present version does not have groups yet. It will be improved in future
+	 * pull requests. 
+	 */
+	public static FrankDocModel populate() {
+		FrankDocModel result = new FrankDocModel();
+		result.createConfigChildDescriptorsFrom(DIGESTER_RULES);
+		result.findOrCreateElementType(Utils.getClass("nl.nn.adapterframework.core.IAdapter"));
+		return result;
+	}
 
 	public FrankDocModel() {
+		configChildDescriptors = new HashMap<>();
 		groups = new ArrayList<>();
 	}
 
+	public void createConfigChildDescriptorsFrom(String path) {
+		Resource resource = Resource.getResource(path);
+		if(resource == null) {
+			throw new RuntimeException(String.format("Cannot find resource on the classpath: [%s]", path));
+		}
+		try {
+			XmlUtils.parseXml(resource.asInputSource(), new Handler(path));
+		}
+		catch(IOException e) {
+			throw new RuntimeException(String.format("An IOException occurred while parsing XML from [%s]", path), e);
```

Comment:
Je moet hier, en op andere plekken, niet zomaar RuntimeExceptions gooien, maar 'normale' (checked) exceptions, die je vangt en afhandelt op een plek waar je er wat mee kan.

Reply:
Ik heb alle runtime exceptions uit de "nieuwe" code weggehaald. In de oude Frank!Doc code kwamen al runtime exceptions voor. Die heb ik laten staan. In test code heb ik de runtime exceptions ook laten staan.